From e378f3187b1f72ad89a2585c6fc273df3477e2f6 Mon Sep 17 00:00:00 2001
From: Yuyao Lin <linyuyao1@huawei.com>
Date: Tue, 17 Jan 2023 21:02:55 +0800
Subject: [PATCH 025/905] [Huawei] USB: Fix kABI for
 usb_device->reset_in_progress

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188188
CVE: CVE-2022-4662

--------------------------------

Fix kABI by using KABI_FILL_HOLE() for usb_device->reset_in_progress.

Signed-off-by: Yuyao Lin <linyuyao1@huawei.com>
---
 include/linux/usb.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/usb.h b/include/linux/usb.h
index a093667991bb..b174ce7260b9 100644
--- a/include/linux/usb.h
+++ b/include/linux/usb.h
@@ -666,7 +666,6 @@ struct usb_device {
 
 	unsigned can_submit:1;
 	unsigned persist_enabled:1;
-	unsigned reset_in_progress:1;
 	unsigned have_langid:1;
 	unsigned authorized:1;
 	unsigned authenticated:1;
@@ -678,6 +677,7 @@ struct usb_device {
 	unsigned usb2_hw_lpm_allowed:1;
 	unsigned usb3_lpm_u1_enabled:1;
 	unsigned usb3_lpm_u2_enabled:1;
+	KABI_FILL_HOLE(unsigned reset_in_progress:1)
 	int string_langid;
 
 	/* static strings from the device */
-- 
2.17.1

